src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -37,22 +37,25 @@
 
 ;;;; Handlers
 
-(defn choose-address-and-name [db address name]
-  (update db :wallet/send-transaction assoc :to-address address :to-name name))
+(defn update-address-and-name [cofx address name]
+  (update-in cofx [:db :wallet/send-transaction] assoc :to-address address :to-name name))
 
 (handlers/register-handler-fx
   :choose-recipient
-  (fn [{:keys [db]} [_ address name]]
-    (let [{:keys [view-id]} db]
-      (cond-> {:db (choose-address-and-name db address name)}
-        (= :choose-recipient view-id) (assoc :dispatch [:navigate-back])))))
+  (fn [{{:keys [web3] :as db} :db} [_ address name]]
+    (let [{:keys [view-id]} db
+          valid-address? (.isAddress web3 address)]
+      (cond-> {:db db}
```

Comment:
this could be achieved with a let binding for new-db without the need to return a constructed map